pkg/object/httpserver/mux.go

```diff
@@ -78,18 +78,28 @@ type (
 		ipFilter      *ipfilter.IPFilter
 		ipFilterChain *ipfilter.IPFilters
 
-		path          string
-		pathPrefix    string
-		pathRegexp    string
-		pathRE        *regexp.Regexp
-		methods       []string
-		rewriteTarget string
-		backend       string
-		headers       []*Header
+		path            string
+		pathPrefix      string
+		pathRegexp      string
+		pathRE          *regexp.Regexp
+		methods         []string
+		rewriteTarget   string
+		backend         string
+		headers         []*Header
+		matchedPathType PathType
 	}
 
 	// SearchResult is returned by SearchPath
 	SearchResult string
+
+	// PathType is the type of path
+	PathType string
+)
+
+const (
+	PATH   PathType = "path"
```

Comment:
nit: please comment PATH, PREFIX and REGEXP

Reply:
I propose not to introduce the `PathType`, it brings extra complexity and only a few benefits.
The `handleRewrite` function could be changed to below and other code could be changes accordingly:
```go
func (mp *MuxPath) handleRewrite(req context.HTTPRequest) {
	if mp.rewriteTarget == "" {
		return
	}

	path := req.Path()

	if mp.path != "" && mp.path == path {
		req.SetPath(mp.rewriteTarget)
		return
	}
	
	if mp.pathPrefix != "" && strings.HasPrefix(path, mp.pathPrefix) {
		path = mp.pathPrefix + path[len(mp.pathPrefix):]
		req.SetPath(path)
		return
	}
	
	// If we get here, then we are sure the following is true:
	//    mp.pathRE != nil && mp.pathRE.MatchString(path)
	path = mp.pathRE.ReplaceAllString(path, mp.rewriteTarget)
	req.SetPath(path)
}
```